src/ansi-c/c_typecheck_code.cpp

```diff
@@ -903,46 +922,90 @@ void c_typecheck_baset::typecheck_spec_assigns_condition(exprt &condition)
     throw 0;
```

Comment:
Are we trying to get away from "throw 0" to throwing catchable things? Is this an opportunity to fix this one?